src/main/webapp/3Dscript.js

```diff
@@ -45,146 +76,273 @@ function initThreeJs() {
-  plotOrientation();
+  const xCylinder = new THREE.InstancedMesh(geometry, material, pose.length);
+  xCylinder.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(xCylinder);
+
+  return [xCylinder, yCylinder, zCylinder];
+}
+/**
+  * Converts latitude, longitude, and altitude coordinates (LLA) to
+  * Earth-Centered, Earth-Fixed coordinates (ECEF).
+  * @param {number} lat Latitude represented in Decimal Degrees
+  * @param {number} lng Longitude represented in Decimal Degrees
+  * @param {number} alt Altitude represented in Decimal Degrees
+  * @return {array} An array containing x, y, z ECEF coordinates
+  *     all of which are in meters.
+  */
+function llaToEcef(lat, lng, alt) {
+  // LLA to ECEF Paper: https://microem.ru/files/2012/08/GPS.G1-X-00006.pdf
+  const majorAxis = 6378137; // Meters
+  const flatConstant = 1 / 298.25722356; // Scalar
+  const minorAxis = majorAxis*(1-flatConstant); // Meters
+
+  // Eccentricity.
+  const eSq = (Math.pow(majorAxis, 2) -
+    Math.pow(minorAxis, 2)) / Math.pow(majorAxis, 2);
+  const lambda = THREE.Math.degToRad(lat);
+  const phi = THREE.Math.degToRad(lng);
+
+  // Radius of curvature on the Earth
+  const radius = majorAxis /
+    Math.sqrt(1 - eSq * Math.pow(Math.sin(lambda), 2));
+  // Conversion equations.
+  const x = (radius + alt) * Math.cos(lambda) * Math.cos(phi);
+  const y = (radius + alt) * Math.cos(lambda) * Math.sin(phi);
+  const z = ((Math.pow(minorAxis, 2) * radius) /
+    Math.pow(majorAxis, 2) + alt) * Math.sin(lambda);
+
+  return [x, y, z];
 }
 
 /**
- * This uses the pose data to create a blue line representing the pose
- * trajectory. 
+ * Converts Earth-Centered, Earth-Fixed coordinates (ECEF) to
+ * East-North-Up coordinates (ENU).
+ * @param {Object} ecefPose The pose data point being converted to ENU.
+ * @param {Object} ecefOrigin Reference ECEF position, used as the origin
+ *     of the 3D world.
+ * @param {Object} origin Reference LLA position, used as the origin
+ *     of the 3D world.
+ * @return {array} An array containing x, y, z ENU coordinates.
  */
-function plotTrajectory() {
+function ecefToEnu(ecefPose, ecefOrigin, origin) {
+  // ECEF to ENU equation: https://www.mathworks.com/help/map/ref/ecef2enu.html
+  // Localizes the new xyz coordinate using the reference point.
+  const xd = ecefPose.getX() - ecefOrigin.getX();
+  const yd = ecefPose.getY() - ecefOrigin.getY();
+  const zd = ecefPose.getZ() - ecefOrigin.getZ();
+
+  const sinLambda = Math.sin(THREE.Math.degToRad(origin.getX()));
+  const cosLambda = Math.cos(THREE.Math.degToRad(origin.getX()));
+  const cosPhi = Math.cos(THREE.Math.degToRad(origin.getY()));
+  const sinPhi = Math.sin(THREE.Math.degToRad(origin.getY()));
+
+  // Matrix multiplication.
+  const xEast = -sinPhi * xd + cosPhi * yd;
+  const yNorth = -cosPhi * sinLambda * xd - sinLambda *
+    sinPhi * yd + cosLambda * zd;
+  const zUp = cosLambda * cosPhi * xd + cosLambda *
+    sinPhi * yd + sinLambda * zd;
+
```

Comment:
Could you add a link to the math reference?

Reply:
https://www.mathworks.com/help/map/ref/ecef2enu.html, this link is given above on line 230.